cloud/pkg/synccontroller/objectsync.go

```diff
@@ -18,181 +20,73 @@ import (
-
-	nodeName := getNodeName(sync.Name)
-	if pod != nil {
-		syncObjUID := getObjectUID(sync.Name)
-		if syncObjUID != string(pod.GetUID()) {
-			err = apierrors.NewNotFound(schema.GroupResource{
-				Group:    "",
-				Resource: "pods",
-			}, sync.Spec.ObjectName)
-		}
-	}
-
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			pod = &v1.Pod{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      sync.Spec.ObjectName,
-					Namespace: sync.Namespace,
-					UID:       types.UID(getObjectUID(sync.Name)),
-				},
-			}
-		} else {
-			klog.Errorf("Failed to manage pod sync of %s in namespace %s: %v", sync.Name, sync.Namespace, err)
-			return
-		}
-	}
-	sendEvents(err, nodeName, sync, model.ResourceTypePod, pod.ResourceVersion, pod)
+var forceSet = map[string]string{
+	"pod":       "pods",
+	"configmap": "configmaps",
+	"secret":    "secrets",
+	"service":   "services",
+	"endpoint":  "endpoints",
+	"node":      "nodes",
 }
 
-func (sctl *SyncController) manageConfigMap(sync *v1alpha1.ObjectSync) {
-	configmap, err := sctl.configMapLister.ConfigMaps(sync.Namespace).Get(sync.Spec.ObjectName)
-
-	nodeName := getNodeName(sync.Name)
-	if configmap != nil {
-		syncObjUID := getObjectUID(sync.Name)
-		if syncObjUID != string(configmap.GetUID()) {
-			err = apierrors.NewNotFound(schema.GroupResource{
-				Group:    "",
-				Resource: "configmaps",
-			}, sync.Spec.ObjectName)
-		}
-	}
+func (sctl *SyncController) manageObject(sync *v1alpha1.ObjectSync) {
+	var object metav1.Object
 
+	gv, err := schema.ParseGroupVersion(sync.Spec.ObjectAPIVersion)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			configmap = &v1.ConfigMap{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      sync.Spec.ObjectName,
-					Namespace: sync.Namespace,
-					UID:       types.UID(getObjectUID(sync.Name)),
-				},
-			}
-		} else {
-			klog.Errorf("Failed to manage configMap sync of %s in namespace %s: %v", sync.Name, sync.Namespace, err)
-			return
-		}
+		return
 	}
-	sendEvents(err, nodeName, sync, model.ResourceTypeConfigmap, configmap.ResourceVersion, configmap)
-}
-
-func (sctl *SyncController) manageSecret(sync *v1alpha1.ObjectSync) {
-	secret, err := sctl.secretLister.Secrets(sync.Namespace).Get(sync.Spec.ObjectName)
-
-	nodeName := getNodeName(sync.Name)
-
-	if secret != nil {
-		syncObjUID := getObjectUID(sync.Name)
-		if syncObjUID != string(secret.GetUID()) {
-			err = apierrors.NewNotFound(schema.GroupResource{
-				Group:    "",
-				Resource: "secrets",
-			}, sync.Spec.ObjectName)
-		}
+	resource := sync.Spec.ObjectKind // In fact, ObjdctKind is Object resources like "pods"
+	if v, ok := forceSet[resource]; ok {
+		resource = v
 	}
-
+	gvr := gv.WithResource(resource)
+	//ret, err := informers.GetInformersManager().GetDynamicSharedInformerFactory().ForResource(gvr).Lister().ByNamespace(sync.Namespace).Get(sync.Spec.ObjectName)
+	ret, err := sctl.kubeclient.Resource(gvr).Namespace(sync.Namespace).Get(context.TODO(), sync.Spec.ObjectName, metav1.GetOptions{})
```

Comment:
We need to use dynamic in future, right?

Reply:
yeah, will do in later pr.